lib/membrane_rtc_engine/endpoints/webrtc/variant_selector.ex

```diff
@@ -142,6 +142,19 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC.VariantSelector do
     }
   end
 
+  # TODO bytes per second or bits per second
+  @doc """
+  Updates bitrate assigned to a variant in the selector. Expected bitrate unit is bps.
+  """
+  @spec update_variant_bitrate(t(), Track.variant(), non_neg_integer()) :: t()
+  def update_variant_bitrate(%__MODULE__{} = selector, variant, bitrate) do
+    variant_bitrates = %{selector.variant_bitrates | variant => bitrate}
+
+    selector
+    |> Map.put(:variant_bitrates, variant_bitrates)
+    |> manage_allocation()
```

Comment:
You can just...
```suggestion
    selector
    |> put_in([:variant_bitrates, variant], bitrate)
    |> manage_allocation()
```

Reply:
AFAIK structs does not implement Access behaviour which is required to use `put_in`, am I missing something?

Reply:
Ah, true. I forgot that it's a struct. But FYI, for future reference: https://hexdocs.pm/bunch/Bunch.Access.html